tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Logging/ILogger/DirectSubmission/DirectSubmissionLoggerProvider.cs

```diff
@@ -40,10 +38,20 @@ internal DirectSubmissionLoggerProvider(
             IExternalScopeProvider? scopeProvider)
         {
             _sink = sink;
-            _formatter = formatter;
             _minimumLogLevel = minimumLogLevel;
             _createLoggerFunc = CreateLoggerImplementation;
-            _scopeProvider = scopeProvider;
+
+#if NETCOREAPP3_1_OR_GREATER
```

Comment:
This is kind of hard to follow with the `#endif` trying to come right after the else. Can you do rewrite this into full `#if NETCOREAPP3_1_OR_GREATER`, `#else`, and `#endif` blocks, with code duplication as needed?